multiwallet.go

```diff
@@ -0,0 +1,434 @@
+package dcrlibwallet
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/asdine/storm"
+	"github.com/asdine/storm/q"
+	"github.com/decred/dcrwallet/errors/v2"
+	wallet "github.com/decred/dcrwallet/wallet/v3"
+	"github.com/raedahgroup/dcrlibwallet/internal/netparams"
+	"github.com/raedahgroup/dcrlibwallet/utils"
+	bolt "go.etcd.io/bbolt"
+)
+
+const (
+	logFileName   = "dcrlibwallet.log"
+	walletsDbName = "wallets.db"
+)
+
+type MultiWallet struct {
+	dbDriver string
+	rootDir  string
+	db       *storm.DB
+	configDB *storm.DB
+
+	activeNet *netparams.Params
+	wallets   map[int]*LibWallet
+	*syncData
+
+	shuttingDown chan bool
+	cancelFuncs  []context.CancelFunc
+}
+
+func NewMultiWallet(rootDir, dbDriver, netType string) (*MultiWallet, error) {
+	rootDir = filepath.Join(rootDir, netType)
+	initLogRotator(filepath.Join(rootDir, logFileName))
+	errors.Separator = ":: "
+
+	activeNet := utils.NetParams(netType)
+	if activeNet == nil {
+		return nil, errors.E("unsupported network type: %s", netType)
+	}
+
+	err := os.MkdirAll(rootDir, 0700)
+	if err != nil {
+		return nil, errors.E("failed to create wallet db directory: %v", err)
+	}
+
+	configDbPath := filepath.Join(rootDir, userConfigDbFilename)
+	configDB, err := storm.Open(configDbPath)
+	if err != nil {
+		if err == bolt.ErrTimeout {
+			// timeout error occurs if storm fails to acquire a lock on the database file
+			return nil, errors.E("settings db is in use by another process")
+		}
+		return nil, errors.E("error opening settings db store: %s", err.Error())
+	}
+
+	db, err := storm.Open(filepath.Join(rootDir, walletsDbName))
+	if err != nil {
+		log.Errorf("Error opening wallet database: %s", err.Error())
+		if err == bolt.ErrTimeout {
+			// timeout error occurs if storm fails to acquire a lock on the database file
+			return nil, errors.E("wallet database is in use by another process")
+		}
+		return nil, errors.E("error opening wallet index database: %s", err.Error())
+	}
+
+	// init database for saving/reading wallet objects
+	err = db.Init(&LibWallet{})
```

Comment:
In my previous review recommending to save `WalletProperties` rather than `LibWallet`s, this is what I actually had in mind.